File: backend/EmengencyMS.API/Presentation/Contracts/Emergency/CreateEmergency.cs
using Microsoft.AspNetCore.Http;
using Presentation.Contracts.Location;
using Presentation.Contracts.Source;
using Presentation.Contracts.Street;
using System.ComponentModel.DataAnnotations;

namespace Presentation.Contracts.Emergency;

public class CreateEmergency
{
    public string? Title { get; set; }
    public string? Description { get; set; }
    public int EmergencyType { get; set; }
    public int? EmergencySubType { get; set; }
    public DateTime AccidentDate { get; set; }
    [Range(1, 10)]
    public int? Severity { get; set; }
    public int? Casualties { get; set; }
    public int? Injured { get; set; }
    public decimal? EconomicLoss { get; set; }
    public double? Duration { get; set; }
    public CreateLocation Location { get; set; }
    public CreateSource Source { get; set; }
    public CreateStreet Street { get; set; }
    public IList<IFormFile> Images { get; set; }

}

File: frontend/emergencyMS-web/src/app/requests/CreateEmergency.ts
import { Image } from "../models/Image";
import { CreateLocation } from "./CreateLocation";
import { CreateSource } from "./CreateSource";
import { CreateStreet } from "./CreateStreet";

export interface CreateEmergency {
    title?: string;
    description?: string;
    emergencyType: number;
    emergencySubType?: number;
    accidentDate: Date;
    severity?: number;
    casualties?: number;
    injured?: number;
    economicLoss?: number;
    duration?: number;
    location: CreateLocation;
    source: CreateSource;
    street: CreateStreet;
    images: File[];
    imagesEntities?: Image[];
    imagesToDelete?: string[];
}